src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1107,7 +1107,9 @@ var ReactCompositeComponent = {
     if (__DEV__) {
       var componentName = component && component.getName ?
         component.getName() : 'a component';
-      warning(publicComponentInstance != null,
+      warning(
+        publicComponentInstance != null ||
+        component._compositeType !== CompositeTypes.StatelessFunctional,
```

Comment:
Right, it would. I avoid polymorphic accesses like this whenever possible because they're slower but I'm not sure there's a better option here.